org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -352,6 +355,8 @@ protected void addProjectSourceFolders(IClasspathDescriptor classpath, Map<Strin
           } catch(CoreException ex) {
             log.error("Failed to determine compiler test exclusions, assuming defaults", ex);
           }
+          isTestCompilationSkipped = "true" //$NON-NLS-1$
+              .equals(maven.getMojoParameterValue(mavenProject, compile, "skip", String.class, monitor)); //$NON-NLS-1$
```

Comment:
```suggestion
          isTestCompilationSkipped = Boolean.TRUE
              .equals(maven.getMojoParameterValue(mavenProject, compile, "skip", Boolean.class, monitor)); //$NON-NLS-1$
```
The skip field of the compiler-plugin is of type `boolean`:
https://github.com/apache/maven-compiler-plugin/blob/b225b1c29b1ee2fc2b178d2f2f1b920afe1d9a75/src/main/java/org/apache/maven/plugin/compiler/TestCompilerMojo.java#L67-L68

Therefore I think it is better to let maven handle the value conversion.